Source/SwiftLintFramework/Rules/Style/SortedEnumCasesRule.swift

```diff
@@ -0,0 +1,95 @@
+import Foundation
+import SwiftSyntax
+
+struct SortedEnumCasesRule: ConfigurationProviderRule, SwiftSyntaxRule, OptInRule {
+    var configuration = SeverityConfiguration(.warning)
+
+    init() {}
+
+    static let description = RuleDescription(
+        identifier: "sorted_enum_cases",
+        name: "Sorted Enum Cases",
+        description: "Enum cases should be sorted",
+        kind: .style,
+        nonTriggeringExamples: [
+            Example("""
+            enum foo {
+                case a
+                case b
+                case c
+            }
+            """),
+            Example("""
+            enum foo {
+                case a, b, c
+            }
+            """),
+            Example("""
+            enum foo {
+                case a
+                case b, c
+            }
+            """)
+        ],
+        triggeringExamples: [
+            Example("""
+            enum foo {
+                ↓case b
+                ↓case a
+                case c
+            }
+            """),
+            Example("""
+            enum foo {
+                case ↓b, ↓a, c
+            }
+            """),
+            Example("""
+            enum foo {
+                ↓case b, c
+                ↓case a
+            }
+            """),
+            Example("""
+            enum foo {
+                case a
+                case b, ↓d, ↓c
+            }
+            """)
+        ]
+    )
+
+    func makeVisitor(file: SwiftLintFile) -> ViolationsSyntaxVisitor {
+        Visitor(viewMode: .sourceAccurate)
+    }
+}
+
+private extension SortedEnumCasesRule {
+    final class Visitor: ViolationsSyntaxVisitor {
+        override var skippableDeclarations: [DeclSyntaxProtocol.Type] {
+            return [VariableDeclSyntax.self]
+        }
```

Comment:
You probably rather want:
```suggestion
        override var skippableDeclarations: [DeclSyntaxProtocol.Type] { .allExcept(EnumDeclSyntax.self) }
```
That way, it skips protocols, classes, structs, ... but not enums.